calcom/cal.com
# fix: event Duration displays incorrectly in event name

## What does this PR do?



- Fixes #16887
 
[Loom video of the fix](https://www.loom.com/share/0d01be1f663f45e3920645e2a503d41c?sid=c5c0ac1b-7dec-43c6-b9be-b6aa49b63389)

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

Steps to test the fix (see loom video)
1. Create an event with the following attributes
1.1. event has multiple duration
1.2. booker is allowed to select duration
1.3 under advanced tab of the event the `with event name` field is setup something like this `Event Name {Event duration}: {Organiser} and {Scheduler}`
2. Book this event with any thing but the default duration
3. Check `baseaddress/booking/{bookingid}` 
4. The name of the event **should include the selected duration**
5. The email **should include the selected duration**

___

- Are there environment variables that should be set ?  **no**
- What are the minimal test data to have? A regular user and an event as described above
- What is expected (happy path) to have (input and output)? ** name of the event is in line with selected non default duration everywhere**
- Any other important info that could help to test that PR ? **no**

## Checklist

## What caused the bug 

`getEventName` function in `./packages/core/event.ts` was computing event name based on the default length of the event

## How was the issue fixed ?
- `getEventName` was updated to accept two additional optional parameters (start time and end time) and if these parameters were passed then the duration in the event name is computed from these two additional parameters instead of using the default value

- function calls to the above function were updated to pass the two new arguments

## Environment

production
